common/transport/amqp/src/amqp_message.ts

```diff
@@ -141,47 +144,59 @@ export class AmqpMessage {
       throw new ReferenceError('amqpMessage cannot be \'' + amqpMessage + '\'');
     }
 
-    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_009: [The `toMessage` method shall set the `Message.data` of the message to the content of the `AmqpMessage.body` property.]*/
-    let msg = new Message(amqpMessage.body);
+    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_009: [The `toMessage` method shall set the `Message.data` of the message to the content of the `AmqpMessage.body.content` property.]*/
+    let msg: Message = ( amqpMessage.body ) ? ( new Message(amqpMessage.body.content) ) : ( new Message(undefined) );
 
-    if (amqpMessage.properties) {
-      /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_005: [The `toMessage` method shall set the `Message.to` property to the `AmqpMessage.properties.to` value if it is present.]*/
-      if (amqpMessage.properties.to) {
-        msg.to = amqpMessage.properties.to;
-      }
+    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_005: [The `toMessage` method shall set the `Message.to` property to the `AmqpMessage.to` value if it is present.]*/
+    if (amqpMessage.to) {
+      msg.to = amqpMessage.to;
+    }
 
-      /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_006: [The `toMessage` method shall set the `Message.expiryTimeUtc` property to the `AmqpMessage.properties.absoluteExpiryTime` value if it is present.]*/
-      if (amqpMessage.properties.absoluteExpiryTime) {
-        msg.expiryTimeUtc = amqpMessage.properties.absoluteExpiryTime;
-      }
+    /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_006: [The `toMessage` method shall set the `Message.expiryTimeUtc` property to the `AmqpMessage.absolute_expiry_time` value if it is present.]*/
+    if (amqpMessage.absolute_expiry_time) {
+      msg.expiryTimeUtc = amqpMessage.absolute_expiry_time;
+    }
 
-      /*Codes_SRS_NODE_IOTHUB_AMQPMSG_16_004: [The `toMessage` method shall set the `Message.messageId` property to the `AmqpMessage.properties.messageId` value if it is present.]*/
-      if (amqpMessage.properties.messageId) {
-        msg.messageId = amqpMessage.properties.messageId;
+    //
+    // SERIALIZATION DEPENDENCY!!
+    // The rhea library will de-serialize an encoded uuid (0x98) as a 16 byte buffer.
```

Comment:
>(0x98) [](start = 58, length = 6)

Silly question, and ignore me if this is dumb, but is this 0x98 type identifier available to you here, or has it already been discarded at this point?  (I assume it's been discarded otherwise you wouldn't have written the code this way, but I still have to ask).

Reply:
You are very smart.

---
In reply to: [209371716](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209371716) [](ancestors = 209371716)